databases/week2/task4.sql

```diff
@@ -0,0 +1,59 @@
+CREATE DATABASE smoothie;
+
+USE smoothie;
+
+CREATE TABLE creator (
+	creator_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,age_range ENUM
+	,Gender ENUM
+	,PRIMARY KEY (creator_id)
+	);
+
+CREATE TABLE smoothie (
+	smoothie_id INT NOT NULL
+	,NAME VARCHAR(255)
+	,category VARCHAR(255)
+	,ingredients INT NOT NULL
+	,creator INT NOT NULL
+	,FOREIGN KEY (category) REFERENCES colorType(category)
```

Comment:
Category is the name of the row in a bridge table colorType, I thought I could use it as a foreign key? I will do more research on that topic:)

Reply:
You can but, at first, you need to create this key and after that you can do references to it. Example from JS: you cannot use variable before defining it.

Reply:
So do I need to make it a primary key for its native table first? (category is already declared in the bridge table) :)

Reply:
You need to create table at first where you define your id. Then you can add foreign key whenever you want it (which will be a reference to this id). Because now when fx I tried to run your queries I got an error because in your code you made a reference to the ids which you're defining later. And I need to scroll through your code to find necessary table. But this necessary table could have it's own "undefined" foreign key. And it's a hell circle😅 So I suggest you to move all queries for foreign keys which appeared before their defining and add these queries later using ALTER TABLE. Does it make sense?